src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -703,6 +704,13 @@ internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions,
                 Debug.Assert(explicitInterfaceSpecifier != null);
                 _explicitInterfaceType.CheckAllConstraints(conversions, new SourceLocation(explicitInterfaceSpecifier.Name), diagnostics);
             }
+
+            if (_refKind == RefKind.RefReadOnly)
```

Comment:
>if (_refKind == RefKind.RefReadOnly) [](start = 12, length = 36)

It feels like we should be moving similar checks for all source symbols into an override of AfterAddingTypeMembersChecks API. This should ensure that the checks won't cause circularity issues in the future due to some changes in the compiler's behavior.